### ISSUE TITLE
Why is this called JSON viewer?

### ISSUE THREAD

**Issue by Ghostbird** [hivivo/ngx-json-viewer#145]
If I feed it some data, it will show an expandable view representing that data. What does that have to do with JSON?
In fact if I feed it JSON, it shows a single entry of type string, which is correct, but useless.

Note: I'm just a developer who's slightly confused. A colleague of mine added this component to our software, and it's working as intended for our purpose. However, I was reviewing the code, and it semantically makes no sense, which prompted me to investigate. I found that my colleague inputs non-JSON in an input that's called JSON, and it works. If I "correct" the code and I feed it actual JSON, it doesn't work any more.

In Angular template form, this works, but reads as wrong in a review.

```html
<ngx-json-viewer [json]="data"></ngx-json-viewer>
```

This reads correctly when reviewing, but doesn't actually work.

```html
<ngx-json-viewer [json]="data | json"></ngx-json-viewer>
```

In conclusion, this component is clearly a data viewer, but I'm baffled where the JSON aspect comes in. As far as I can see it really doesn't do anything with JSON specifically. In fact, I personally think it's utterly useless at handling JSON.

**Comment by dan-lugg** [comment]
I mean, sure, it's a JavaScript object data visualizer.

But the data format of most interoperability between things is JSON, whether the origin or recipient system uses JavaScript at all.

JavaScript applications, notably front-end web applications, most often receive JSON data from remote sources.

This let's you visualize that data by dumping it into an Angular component.

I get the argument of semantics, but is there really any value in renaming it to something more obscure, when JSON viewer is pretty ubiquitous, even if it's "incorrect"?

Source: I got here looking for `angular json viewer`.

**Comment by Ghostbird** [comment]
It caused quite a bit of confusion in our team. A developer had added this just to quickly provide a visual data tree. But we were implementing an overarching data viewer that has two different display modes. An expandable tree view, such as this component offers, and an actual formatted JSON view.

You can imagine that it becomes really confusing when the view that shows the JSON contains a JSON viewer, and the view that shows the tree view also contains something named a JSON viewer.

Combined with a less than stellar performance due to the inefficient recursive re-parsing implementation and recursive component instantiation of this component, we decided to write our own data viewer component. It only parses a single time, and generates a DOM‌ tree in a single Angular component. This worked for our use case (100kB-10MB data).

In our case the data isn't necessarily JSON at all. In fact, the real JSON viewer we added stringifies the data to ease manual copying of the data to other systems. However, even if it were, naming things for a replaceable implementation detail on a different abstraction level would be a bad idea. Following the proposed logic, this component should _also_ be called XML viewer, Protobuf viewer, etc.

**Comment by Ghostbird** [comment]
@dan-lugg  You came here looking for `angular json viewer`. Were you actually looking for a JSON viewer, or for a data tree visualiser such as this component? Because in the first case, you didn't find what you _searched_ for. But if you did find what you were _looking_ for, I really don't comprehend how you decided on that search terminology.
